build_tools/fish_xgettext.fish

```diff
@@ -15,18 +19,22 @@ begin
         echo ""
     end
 
-    set -l rust_extraction_file (mktemp)
+    set -g repo_root (status dirname)/..
 
-    # We need to build to ensure that the proc macro for extracting strings runs.
-    FISH_GETTEXT_EXTRACTION_FILE=$rust_extraction_file cargo check
-    or exit 1
+    set -l --erase rust_extraction_file
+    if set -l --query _flag_use_existing_template
+        set rust_extraction_file $_flag_use_existing_template
+    else
+        set rust_extraction_file (mktemp)
+        # We need to build to ensure that the proc macro for extracting strings runs.
+        FISH_GETTEXT_EXTRACTION_FILE=$rust_extraction_file cargo check
+        or exit 1
+    end
 
     # Get rid of duplicates and sort.
     msguniq --no-wrap --strict --sort-output $rust_extraction_file
     or exit 1
 
-    rm $rust_extraction_file
```

Comment:
I think if `--use-existing-template` is not given, we should delete this temporary file as before